lib/iris/coords.py

```diff
@@ -2775,6 +2779,366 @@ def xml_element(self, doc):
+            Use a :class:`numpy.ma.core.MaskedArray` if element lengths vary -
+            mask unused index 'slots' within each element. Use a
+            :class:`dask.array.Array` to keep indices 'lazy'.
+        * cf_role (str):
+            Denotes the topological relationship that this connectivity
+            describes. Made up of this array's elements, and the indexed
+            element within each element. E.g. ``face_node_connectivity``,
+            ``edge_face_connectivity``.
+
+        Kwargs:
+
+        * long_name (str):
+            Descriptive name of the connectivity.
+        * var_name (str):
+            The netCDF variable name for the connectivity.
+        * attributes (dict):
+            A dictionary containing other cf and user-defined attributes.
+        * start_index (int):
+            Either ``0`` or ``1``. Denotes whether :attr:`indices` uses 0- or
+            1-based indexing (allows support for Fortran and legacy NetCDF
+            files).
+        * element_dim (int):
+            Either ``0`` or ``1``. Denotes which dimension of :attr:`indices`
+            refers to the contents of individual elements (allows support
+            for fastest varying index being either first or last). E.g. for
+            ``face_node_connectivity``, where the faces are 3-sided:
+            ``indices.shape[element_dim] = 3``.
+
+        """
+        # Configure the metadata manager.
+        self._metadata_manager = metadata_manager_factory(ConnectivityMetadata)
+
+        self._validate_arg_vs_list("start_index", start_index, [0, 1])
+        # indices array will be 2-dimensional, so must be either 0 or 1.
+        self._validate_arg_vs_list("element_dim", element_dim, [0, 1])
+        # TODO: find a better 'common' location to store valid cf_roles.
+        valid_cf_roles = [
+            "edge_node_connectivity",
+            "face_node_connectivity",
+            "face_edge_connectivity",
+            "face_face_connectivity",
+            "edge_face_connectivity",
+            "boundary_node_connectivity",
+            "volume_node_connectivity",
+            "volume_edge_connectivity",
+            "volume_face_connectivity",
+            "volume_volume_connectivity",
+        ]
+        self._validate_arg_vs_list("cf_role", cf_role, valid_cf_roles)
+
+        self._metadata_manager.start_index = start_index
+        self._metadata_manager.element_dim = element_dim
+        self._metadata_manager.cf_role = cf_role
+
+        super().__init__(
+            values=indices,
+            standard_name=None,
+            long_name=long_name,
+            var_name=var_name,
+            units="1",
+            attributes=attributes,
+        )
+
+    @staticmethod
+    def _validate_arg_vs_list(arg_name, arg, list):
+        if arg not in list:
+            error_msg = (
+                f"Invalid {arg_name} . Got: {arg} . Must be one of: "
+                f"{list} ."
```

Comment:
Curious, but why the spacing before the period (` .`)?

Reply:
I found my original outputs difficult to read; having space either side of the value(s) in focus made them easier to pick out.